docs/DeveloperGuide.md

```diff
@@ -236,71 +236,169 @@ _{Explain here how the data archiving feature will be implemented}_
+| `* *`    | organized person                           | categorise my expenditure                                                                            | better manage specific aspects of my spending                          |
+| `* *`    | new user                                   | use a help command                                                                                   | refer to instructions when I forget how to use the App                 |
+| `* *`    | cautious person                            | set a password for Bamboo                                                                            | keep my expenses private                                               |
+| `* *`    | user with many expenses                    | find expenses via date, keywords, or category                                                        | locate an expense easily                                               |
+| `* *`    | careless user                              | revert my commands                                                                                   | easily undo changes I made to my budgeting                             |
+| `*`      | meticulous user                            | track my saving progress to buy big ticket items                                                     | know how far away am I from the target                                 |
+| `*`      | user who likes to see progress             | use the progress tracker to motivate myself                                                          | keep working at saving up                                              |
+| `*`      | cautious person                            | view my ledger data in a human-readable format and only edit the file when commands are executed     | be assured that the accounts are updated and accurate                  |
+| `*`      | long-time user                             | archive older data from my view                                                                      | manage my expenses easier                                              |
+| `stretch`| user who likes to plan in advance          | simulate future spending                                                                             | visualize my journey towards my financial goals                        |
+| `stretch`| forgetful user                             | receive notifications of budget limits and bill payments                                             | better plan for daily expenditure and make payments on time            | 
 
 *{More to be added}*
 
-### Use cases
+### Use case
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the System is the Bamboo and the Actor is the user, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: Add an expense**
+
+**Preconditions (Needed for v1.2.1)**
```

Comment:
Minor difference. (Needed for v1.2.1) not inside header format in other places

Reply:
Thanks for pointing that out! Tell me if any further changes are needed.